bugbug/repository.py

```diff
@@ -310,27 +310,53 @@ def _hg_log(revs):
     return hg_log(HG, revs)
 
 
-def get_revs(hg, date_from=None):
-    # TODO: Retrieve all commits once calculate_experiences is faster and less memory hungry.
-    if date_from is not None:
-        pushdate = (date_from - relativedelta(months=12)).strftime("%Y-%m-%d")
-        rev_start = f"pushdate('>{pushdate}')"
-    else:
-        rev_start = 0
-
-    print(f"Getting revs from {rev_start} to tip...")
+def get_revs(hg):
+    print(f"Getting revs from 0 to tip...")
 
     args = hglib.util.cmdbuilder(
-        b"log",
-        template="{node}\n",
-        no_merges=True,
-        branch="central",
-        rev=f"{rev_start}:tip",
+        b"log", template="{node}\n", no_merges=True, branch="central", rev=f"0:tip"
     )
     x = hg.rawcommand(args)
     return x.splitlines()
 
 
+class exp_queue:
+    def __init__(self, start_day, maxlen, default):
+        self.list = deque([default] * maxlen, maxlen=maxlen)
+        self.start_day = start_day - (maxlen - 1)
+        self.default = default
+
+    @property
+    def last_day(self):
```

Comment:
nit: you could replace with self.last_day = start_day (in ctor) or = day (in __setitem__)

Reply:
I did that at first, but then I wanted to avoid having an additional variable to think about.